pilot/pkg/model/validation_test.go

```diff
@@ -1746,6 +1746,57 @@ func TestValidateHTTPRoute(t *testing.T) {
 	}
 }
 
+func TestValidateRouteDestination(t *testing.T) {
+	testCases := []struct {
+		name   string
+		routes []*networking.RouteDestination
+		valid  bool
+	}{
+		{name: "simple", routes: []*networking.RouteDestination{&networking.RouteDestination{
+			Destination: &networking.Destination{Host: "foo.baz"},
+		}}, valid: true},
+		{name: "no destination", routes: []*networking.RouteDestination{&networking.RouteDestination{
+			Destination: nil,
+		}}, valid: false},
+		{name: "weighted", routes: []*networking.RouteDestination{&networking.RouteDestination{
+			Destination: &networking.Destination{Host: "foo.baz.south"},
+			Weight:      25,
+		}, &networking.RouteDestination{
+			Destination: &networking.Destination{Host: "foo.baz.east"},
+			Weight:      75,
+		}}, valid: true},
+		{name: "zero weight", routes: []*networking.RouteDestination{&networking.RouteDestination{
+			Destination: &networking.Destination{Host: "foo.baz.south"},
+			Weight:      5,
+		}, &networking.RouteDestination{
+			Destination: &networking.Destination{Host: "foo.baz.east"},
+			Weight:      0,
```

Comment:
why is zero weight not allowed?

Reply:
How would Envoy determine cluster buckets in case of 0 weights?